### PR TITLE
prevent agents which can kick from moving into bomb if bomb cannot mo…

### PR DESCRIPTION
**Pull request by kenjyoung**
…ve due to obstacle

### CODE REVIEW

File: pommerman/forward_model.py [hunk 1]
```diff
@@ -159,7 +159,11 @@ def has_position_conflict(counter):
                     elif not agent.can_kick:
                         agent.stop()
                     else:
-                        next_positions[agent.agent_id] = next_position
+                        invalid_values = list(range(len(constants.Item)+1))[1:]
+                        if not utility.is_valid_direction(curr_board, next_position, action, invalid_values=invalid_values):
+                            agent.stop()
+                        else:
+                            next_positions[agent.agent_id] = next_position
```
Comment (cinjon) [inline pommerman/forward_model.py]: I see. So by the time we get to the else statement, we know that `next_position` is a bomb and the agent can kick. However, if the position _after_ `next_position` is not a passage, then we still don't want to move the agent. Good catch.

Let's do this slightly differently and say:
```
after_next_position = utility.get_next_position(next_position, action)
if not utility.position_on_board(curr_board, after_next_position) or not utility.position_on_passage(curr_board, after_next_position):
    agent.stop()
else:
    ...
```

Sg?

Comment (cinjon) [reply pommerman/forward_model.py]: Want to make that change on this PR?

Comment (kenjyoung) [reply pommerman/forward_model.py]: Sure! Made two slight changes to what you wrote, seems to work now.

**Comment by cinjon** [comment]
lgtm.

**Pull request merged**